### ISSUE TITLE
Window focus callback not called when losing focus on i3wm multimonitor setup

### ISSUE THREAD

**Issue by slowriot** [glfw/glfw#967]
One of our beta testers reports that switching focus away from the program to another monitor with keyboard shortcuts fails to trigger the window focus callback for the window losing focus.  They are using a multimonitor setup with i3wm, with the program running fullscreen on one monitor.

Their bug report to us: https://bitbucket.org/voxelstorm/sphereface/issues/305

They talk about the mouse not being released, because we release it in the callback set with glfwSetWindowFocusCallback - it seems in this case it's not triggered.

Unfortunately I don't have a similar setup here with which I could try to reproduce the issue, so I'm just passing it on as is.

**Comment by elmindreda** [comment]
What does it mean that you release the mouse?

**Comment by slowriot** [comment]
@elmindreda the user is referring to our behaviour of "capturing" / "releasing" the mouse from the window by using `glfwSetInputMode(window_main, GLFW_CURSOR, GLFW_CURSOR_DISABLED);` and `glfwSetInputMode(window_main, GLFW_CURSOR, GLFW_CURSOR_NORMAL);` respectively.  In the case they're describing, these are called after we get a window focus callback, i.e. cursor mode should be returned to GLFW_CURSOR_NORMAL when we detect window focus has been lost.

**Comment by elmindreda** [comment]
I see. GLFW itself should disable all effects of disabled cursor mode when the window loses focus, so there should be no need to do it on your end. What version of GLFW are you using?

**Comment by slowriot** [comment]
Yes, it does, but we switch the game to pause mode when it loses focus, and we don't want to recapture the mouse right away in any case.

We're using GLFW3.3, from the master branch here.

However, the reporter of our issue has just told us he can no longer replicate the problem - it sounds like it may have been a spurious issue with his window manager, in that case, and this issue may be invalid.

**Comment by elmindreda** [comment]
> Yes, it does, but we switch the game to pause mode when it loses focus, and we don't want to recapture the mouse right away in any case.

Sounds like you're doing everything right, then.

> However, the reporter of our issue has just told us he can no longer replicate the problem - it sounds like it may have been a spurious issue with his window manager, in that case, and this issue may be invalid.

All right. I'll try reproducing this on i3wm the next time I'm on X11.
